client/my-sites/domains/domain-management/list/all-domains.jsx

```diff
@@ -409,11 +409,14 @@ class AllDomains extends Component {
 			} );
 		}
 
+		const queryParams = new URLSearchParams( window.location.search );
+
 		return (
 			<>
 				<div className="all-domains__filter">{ this.renderDomainTableFilterButton() }</div>
 				<DomainsTable
 					currentRoute={ currentRoute }
+					defaultSortKey={ queryParams.get( 'sortKey' ) }
```

Comment:
I think we should check that this key is an option. Otherwise, the page will through if someone passes a bad param.

This change to the line to the below will ignore the query parameter if it's incorrect.
```suggestion
					defaultSortKey={ domainsTableColumns?.find(c => c.name === queryParams.get( 'sortKey' ))?.name }
```